api/src/main/java/de/minebench/tresor/services/npc/NPCMetadata.java

```diff
@@ -0,0 +1,31 @@
+package de.minebench.tresor.services.npc;
+
+import de.minebench.tresor.services.npc.skin.NPCSkinData;
+
+public class NPCMetadata {
+
+    private String displayName;
+    private NPCSkinData skinData;
+
+    public NPCMetadata(String displayName, NPCSkinData skinData) {
+        this.displayName = displayName;
+        this.skinData = skinData;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public NPCSkinData getSkinData() {
+        return skinData;
+    }
+
+    public void setDisplayName(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public void setSkinData(NPCSkinData skinData) {
+        this.skinData = skinData;
+    }
+
+}
```

Comment:
The main point of this class is separation of concerns. It also allows you to copy over metadata and apply it to multiple NPCs.

Reply:
Hm, the copying is a good idea but I feel like exposing a separate copy method for all of the NPCs data (even one not exposed by the service but only internal to the implementing plugin) would be better. (And only separating it for two values feels a bit over the top tbh. unless you had the idea to add more than just those two?)

Reply:
The idea is indeed to add more fields, such as glowing status etc.

Reply:
Hm, I guess then it would make sense to keep separate. Not sure if two ways of copying data is good though (one just the Tresor metadata, the other with all internal data if the implementation supports that... not exposing it like this would discourage copying of the data via the metadata which might be good as someone could think it also copies internal metadata)

Reply:
> Hm, I guess then it would make sense to keep separate. Not sure if two ways of copying data is good though (one just the Tresor metadata, the other with all internal data if the implementation supports that... not exposing it like this would discourage copying of the data via the metadata which might be good as someone could think it also copies internal metadata)

Any idea for a suggested fix? I could make this a mutable object with dirty state checking but it'd end up being pretty hacky. Renaming the "setData" method to "updateData" will still allow for copying, Embedding the data in the npc class might work but I'm afraid it'll become a god class and be responsible for too much.

Reply:
I think it would be better to just move everything in to the NPC class. I don't think it risks being a god class because most stuff wouldn't actually be behaviour logic, just "field" getters and setters. (It would also make it easier for plugins to implement the service provider. Adding too many interfaces which one would need to implement to support Tresor might discourage some from supporting it natively especially when the traditional Economy etc. providers only required a single interface to be implemented)

Then a separate `NPC#copy(NPC)` method could be used to have the implementation handle the copying of all of the data (or what the implementation thinks all the data relevant to an NPC is).